### DISCUSSION TITLE
MetaPro tutorial problem

### DISCUSSION THREAD

**Discussion by yeolex** [ParkinsonLab/MetaPro#30] [Q&A]
Hi, please help us. We are running the MetarproTutorial but we don't get it. We suspected the pipeline/Metapro.py has more comands than config_mouse_tutorial.ini because the script python3 pipeline/MetaPro.py -c $config -s $read1 -o $output --tutorial quality recognize all the paths of databases and files, but is doesn't found Labels and others. When we run the pipeline obtain this messages:

CHECKING CONFIG
USING CONFIG /MetaPro_docker_tutorial/config_mouse_tutorial.ini
target_rank found! using: genus
AdapterRemoval_minlength found! using: 30
Show_unclassified found! using: No
bypass_log_name no inner section found. using default bypass_log.txt
debug_stop_flag no inner section found. using default none
num_threads no inner section found. using default 12
taxa_existence_cutoff no inner section found. using default 0.1
DNA_DB_mode no inner section found. using default chocophlan
RPKM_cutoff found! using: 0.01
BWA_cigar_cutoff found! using: 90
BLAT_identity_cutoff found! using: 85
BLAT_length_cutoff found! using: 0.65
BLAT_score_cutoff found! using: 60
DIAMOND_identity_cutoff found! using: 85
DIAMOND_length_cutoff found! using: 0.65
DIAMOND_score_cutoff found! using: 60
BWA_mem_footprint no inner section found. using default 5
BLAT_mem_footprint no inner section found. using default 5
DMD_mem_footprint no inner section found. using default 10
BWA_mem_threshold found! using: 60
BLAT_mem_threshold found! using: 60
DIAMOND_mem_threshold found! using: 60
DETECT_mem_threshold found! using: 60
Infernal_mem_threshold found! using: 60
Barrnap_mem_threshold found! using: 60
BWA_pp_mem_threshold found! using: 60
BLAT_pp_mem_threshold found! using: 60
DIAMOND_pp_mem_threshold found! using: 60
GA_final_merge_mem_threshold no inner section found. using default 5
TA_mem_threshold found! using: 60
repop_mem_threshold no inner section found. using default 50
EC_mem_threshold no inner section found. using default 5
BWA_job_limit found! using: 10
BLAT_job_limit found! using: 10
DIAMOND_job_limit found! using: 10
DETECT_job_limit found! using: 10
Infernal_job_limit found! using: 10
Barrnap_job_limit found! using: 10
BWA_pp_job_limit found! using: 10
BLAT_pp_job_limit found! using: 10
DIAMOND_pp_job_limit found! using: 10
GA_final_merge_job_limit no inner section found. using default 9
TA_job_limit no inner section found. using default 9
repop_job_limit no inner section found. using default 1
EC_job_limit no inner section found. using default 9
Centrifuge_job_limit no inner section found. using default 1
Infernal_job_delay no inner section found. using default 5
Barrnap_job_delay no inner section found. using default 5
BWA_job_delay found! using: 5
BLAT_job_delay found! using: 0.005
DIAMOND_job_delay found! using: 5
DETECT_job_delay no inner section found. using default 5
BWA_pp_job_delay found! using: 5
BLAT_pp_job_delay found! using: 0.05
DIAMOND_pp_job_delay found! using: 5
GA_final_merge_job_delay no inner section found. using default 5
TA_job_delay found! using: 10
repop_job_delay no inner section found. using default 10
EC_job_delay no inner section found. using default 1
keep_all found! using: yes
keep_quality found! using: no
keep_host found! using: no
keep_vector found! using: no
keep_rRNA found! using: no
keep_repop found! using: no
keep_assemble_contigs found! using: no
keep_GA_BWA found! using: no
keep_GA_BLAT found! using: no
keep_GA_DIAMOND found! using: no
keep_GA_final found! using: no
keep_TA found! using: no
keep_EC found! using: no
keep_outputs found! using: no
filter_stringency found! using: high
GA_chunk_size found! using: 50000
EC_chunk_size no inner section found. using default 50000
rRNA_chunk_size found! using: 50000
Labels no section found, using default: quality_filter
Labels no section found, using default: host_filter
Labels no section found, using default: vector_filter
Labels no section found, using default: rRNA_filter
Labels no section found, using default: rRNA_filter_split
Labels no section found, using default: rRNA_filter_convert
Labels no section found, using default: rRNA_filter_barrnap
Labels no section found, using default: rRNA_filter_barrnap_merge
Labels no section found, using default: rRNA_filter_barrnap_pp
Labels no section found, using default: rRNA_filter_infernal
Labels no section found, using default: rRNA_filter_infernal_prep
Labels no section found, using default: rRNA_filter_splitter
Labels no section found, using default: rRNA_filter_post
Labels no section found, using default: duplicate_repopulation
Labels no section found, using default: assemble_contigs
Labels no section found, using default: destroy_contigs
Labels no section found, using default: GA_pre_scan
Labels no section found, using default: GA_split
Labels no section found, using default: GA_BWA
Labels no section found, using default: GA_BWA_pp
Labels no section found, using default: GA_BWA_merge
Labels no section found, using default: GA_BLAT
Labels no section found, using default: GA_BLAT_cleanup
Labels no section found, using default: GA_BLAT_cat
Labels no section found, using default: GA_BLAT_pp
Labels no section found, using default: GA_BLAT_merge
Labels no section found, using default: GA_DMD
Labels no section found, using default: GA_DMD_pp
Labels no section found, using default: GA_final_merge
Labels no section found, using default: taxonomic_annotation
Labels no section found, using default: enzyme_annotation
Labels no section found, using default: enzyme_annotation_detect
Labels no section found, using default: enzyme_annotation_priam
Labels no section found, using default: enzyme_annotation_priam_split
Labels no section found, using default: enzyme_annotation_priam_cat
Labels no section found, using default: enzyme_annotation_DMD
Labels no section found, using default: enzyme_annotation_pp
Labels no section found, using default: outputs
Labels no section found, using default: output_copy_gene_map
Labels no section found, using default: output_clean_ec
Labels no section found, using default: output_copy_taxa
Labels no section found, using default: output_network_generation
Labels no section found, using default: output_unique_hosts_singletons
Labels no section found, using default: output_unique_hosts_pair_1
Labels no section found, using default: output_unique_hosts_pair_2
Labels no section found, using default: output_unique_vectors_singletons
Labels no section found, using default: output_unique_vectors_pair_1
Labels no section found, using default: output_unique_vectors_pair_2
Labels no section found, using default: output_combine_hosts
Labels no section found, using default: output_per_read_scores
Labels no section found, using default: output_contig_stats
Labels no section found, using default: output_ec_heatmap
Labels no section found, using default: output_taxa_groupby
Labels no section found, using default: output_read_count
UniVec_Core found! using: /MetaPro_docker_tutorial/databases//UniVec_Core.fasta
Adapter found! using: /MetaPro_docker_tutorial/databases//TruSeq3-PE-2.fa
Host found! using: /MetaPro_docker_tutorial/databases//Mouse_cds.fasta
Rfam found! using: /MetaPro_docker_tutorial/databases//Rfam/Rfam.cm
DNA_DB no inner section found. using default /project/j/jparkin/Lab_Databases/ChocoPhlAn/ChocoPhlAn.fasta
source_taxa_db no inner section found. using default /project/j/jparkin/Lab_Databases/family_llbs
Prot_DB no inner section found. using default /project/j/jparkin/Lab_Databases/nr/nr
Prot_DB_reads no inner section found. using default /project/j/jparkin/Lab_Databases/nr/nr
accession2taxid no inner section found. using default /project/j/jparkin/Lab_Databases/accession2taxid/accession2taxid
nodes found! using: /MetaPro_docker_tutorial/databases//nodes.dmp
names found! using: /MetaPro_docker_tutorial/databases//names.dmp
Kaiju_db no inner section found. using default /project/j/jparkin/Lab_Databases/kaiju_db/kaiju_db_nr.fmi
Centrifuge_db no inner section found. using default /project/j/jparkin/Lab_Databases/centrifuge_db/nt
SWISS_PROT no inner section found. using default /project/j/jparkin/Lab_Databases/swiss_prot_db/swiss_prot_db
SWISS_PROT_map no inner section found. using default /project/j/jparkin/Lab_Databases/swiss_prot_db/SwissProt_EC_Mapping.tsv
PriamDB no inner section found. using default /project/j/jparkin/Lab_Databases/PRIAM_db/
DetectDB no inner section found. using default /project/j/jparkin/Lab_Databases/DETECTv2
WEVOTEDB no inner section found. using default /project/j/jparkin/Lab_Databases/WEVOTE_db/
EC_pathway no inner section found. using default /project/j/jparkin/Lab_Databases/EC_pathway.txt
path_to_superpath found! using: /pipeline/custom_databases/pathway_to_superpathway.csv
MetaGeneMark_model found! using: /pipeline_tools/mgm/MetaGeneMark_v1.mod
enzyme_db found! using: /pipeline/custom_databases/FREQ_EC_pairs_3_mai_2020.txt
taxid_tree no inner section found. using default /pipeline/custom_databases/taxid_trees/family_tree.tsv
kraken2_db no inner section found. using default /pipeline/custom_databases/kraken2_db
dir name: /project/j/jparkin/Lab_Databases/nr
file name: nr
/project/j/jparkin/Lab_Databases/nr/nr
file does not exists

We now that are so many steps with a # , so we think that some errors like nr files does not exists would not matter.

thank's

**Comment by billylaw8** [comment]
1) metapro tutorial isn't meant to be used to process data.  You use main metapro.
2) The intention was to give new bioinformaticians an introduction to data processing at the Canadian bioinformatics workshop.

If you're looking to use the tutorial to process data, don't.
Use main metapro instead.

**Comment by yeolex** [comment]
We used the data obtained from the tutorial (mouse) and did not use any other databases to run the tutorial. However, we finally realized that the images we used were not correct for the tutorial purpose (metapro:test) (email) . We could not find the Docker image described in the tutorial (metapro:2.1.1) and mistakenly assumed that metapro:test was the correct image. We will now try to run the main metapro and update you on our progress. Thanks
